Add Python 3.13 to CI and to release

IvanIsCoding opened the pull request:
<!--
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [x] I ran rustfmt locally
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
-->

This adds Python 3.13 to CI and bumps `cibuildwheel` to the latest version.

I will test the wheel build on my personal repository before merging.

IvanIsCoding commented on noxfile.py:
```diff
@@ -24,7 +24,7 @@
 def install_rustworkx(session):
     session.install(*deps)
-    session.install(".", "-c", "constraints.txt")
+    session.install(".[all]", "-c", "constraints.txt")
```
I removed this by accident in #1290

IvanIsCoding commented:
Putting this on hold until October 7

IvanIsCoding commented:
And we are affected by https://github.com/python-pillow/Pillow/issues/6676, MyPy is getting smarter although not when I wanted

mtreinish reviewed (commented):
This LGTM, but before merging we probably should update the trove classifiers in the setup.py: https://github.com/Qiskit/rustworkx/blob/main/setup.py#L72

IvanIsCoding commented:
> This LGTM, but before merging we probably should update the trove classifiers in the setup.py: https://github.com/Qiskit/rustworkx/blob/main/setup.py#L72

I added 3.13 to that list. I also just removed `win32` from the wheels being built because of #1293. I will test the wheels build in my repository before merging.

IvanIsCoding commented:
The test build passed, all the failures are permission issues because we omitted the PyPI tokens. This is good to go

IvanIsCoding commented:
After the NumPy bump in #1305 we no longer need to drop the Windows 32-bit build. All builds are working now
